Mathlib/Tactic/Linter/DeprecatedDates.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison
+-/
+import Lean.Elab.Command
+import Batteries.Data.String.Matcher
+import Mathlib.Tactic.Linter.Header
+
+/-!
+# The "deprecatedDates" linter
+
+The "deprecatedDates" linter emits a warning when consecutive deprecated declarations
+have different dates but are not separated by a blank line.
+
+When multiple declarations are deprecated, they should either:
+- Have the same deprecation date (if they were deprecated together), or
+- Be separated by a blank line (if they were deprecated at different times).
+
+This helps maintain readability and makes the history of deprecations clearer.
+-/
+
+open Lean Elab Command Linter
+
+namespace Mathlib.Linter
+
+/-- The "deprecatedDates" linter emits a warning when consecutive deprecated declarations
+have different dates but are not separated by a blank line. -/
+register_option linter.style.deprecatedDates : Bool := {
+  defValue := false
+  descr := "enable the deprecatedDates linter"
+}
+
+namespace Style.DeprecatedDates
+
+/-- Check if there's a blank line between two positions in the source. -/
+def hasBlankLineBetween (fm : FileMap) (endPos : String.Pos) (startPos : String.Pos) : Bool :=
+  if endPos ≥ startPos then
+    false
+  else
+    let substring := { str := fm.source, startPos := endPos, stopPos := startPos : Substring }
+    -- Split into lines and check if any line is empty (after trimming)
+    substring.toString.split (· == '\n') |>.any (·.trim.isEmpty)
+
+@[inherit_doc Mathlib.Linter.linter.style.deprecatedDates]
+def deprecatedDatesLinter : Linter where
+  run := withSetOptionIn fun stx ↦ do
+    unless getLinterValue linter.style.deprecatedDates (← getLinterOptions) do
+      return
+    if (← get).messages.hasErrors then
+      return
+
+    -- Only check declaration commands
+    unless [``Lean.Parser.Command.declaration, `lemma].contains stx.getKind do
```

Comment:
This misses out on `deprecated alias`, whose kind is `` `Batteries.Tactic.Alias.alias``.  However, simply adding this does not "just work", since, for instance, the extraction of the declId later does not work for `alias`.